internal/gir/types/types.go

```diff
@@ -40,12 +41,12 @@ var convList = map[string]string{
 	"guintptr": "uintptr",
 	"utf8":     "string",
 	"filename": "string",
+	"GType":    "int",
```

Comment:
was about to merge until I realised this broke my program. I know "broke" is vague, but now sometimes it crashes or sometimes signal callbacks are refused to be called (or some other thing is happening). Need to track down what exactly. But my feeling says this should be an int32

Reply:
https://github.com/GNOME/glib/blob/7d19892f660f733420331c249de32a0435c26599/gobject/gtype.h#L424-L430

it depends on the compiler/settings/platform lol!

https://github.com/GNOME/glib/blob/7d19892f660f733420331c249de32a0435c26599/glib/glibconfig.h.in#L82

https://github.com/GNOME/glib/blob/7d19892f660f733420331c249de32a0435c26599/meson.build#L1713-L1743

Reply:
found the culprit. It is the fact that now the gobject.Value struct is misaligned (it contains GType and data:)
```
    <record name="Value"
            c:type="GValue"
            glib:type-name="GValue"
            glib:get-type="g_value_get_type"
            c:symbol-prefix="value">
      <doc xml:space="preserve"
           filename="gvalue.h"
           line="97">An opaque structure used to hold different types of values.

The data within the structure has protected scope: it is accessible only
to functions within a #GTypeValueTable structure, or implementations of
the g_value_*() API. That is, code portions which implement new fundamental
types.

#GValue users cannot make any assumptions about how data is stored
within the 2 element @data union, and the @g_type member should
only be accessed through the G_VALUE_TYPE() macro.</doc>
      <source-position filename="gvalue.h" line="128"/>
      <field name="g_type" readable="0" private="1">
        <type name="GType" c:type="GType"/>
      </field>
      <field name="data" writable="1">
        <array zero-terminated="0" fixed-size="2">
          <type name="_Value__data__union"/>
        </array>
      </field>
```

See also https://github.com/GNOME/glib/blob/f09bd17e329cca2be2a34594cb7eda99009a1dff/gobject/gvalue.h#L113-L130

on my system, sizeof(GType) gives 8 bytes, whereas sizeof(GValue) gives 24 bytes. Now with this pr it becomes:

type Value struct {
	GType int

	Data uintptr
}

before it was 

type Value struct {
	GType []interface{}
	Data uintptr
}

which is also incorrect, but []interface{} is 24 bytes so we were maybe lucky with alignment. Data should be 16 bytes I guess instead of uintptr

Reply:
Oof, my apologies, this was a foolish assumption on my part.

I'm no longer using APIs that require GType values (was required for some of the gio DBUS API IIRC, but I gave up on that and went back to godbus and hand-rolled the extra stuff that gio does)

So perhaps we just drop this commit for now.